### ISSUE TITLE
Change `unverified` status to `unknown`

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#6714]
`Unverified` terminology is a little too negative / scary. Instead we can just leave the verification status out until the client is verified (keeping pre-existing behavior).

If a user needs verification, they will know to look for the Verification status badge.

**Comment by AndrewDryga** [comment]
@jamilbk what the UX will be if an admin is going to a client trying to see if client is verified? He needs to remember that the row in a vertical table will disappear?

**Comment by jamilbk** [comment]
Hm, I think in the table views we just leave the badge off currently, right?

In the details/show, instead of showing in the vertical table maybe we should just show the same icon/badge?

The reason I opened this issue was because of confusion on a sales call. If we show unverified for every client, the admin was asking, "do I need to manually verify each client?" and "what happens if I don't?". I had to explain that it basically means a rogue device could authenticate and connect to your Resources.

Maybe we can call the field "device verification" or "hardware verification" instead, and currently we only support the "manual" verification method. In the future we could support "certificate" or "mdm" for example. If the device is not verified, we could just say "verification status: unknown", which might be better language to use.

I'd be curious to get @bmanifold's opinion here.

**Comment by jamilbk** [comment]
> what the UX will be if an admin is going to a client trying to see if client is verified?

When an admin needs this feature, they'll look for it. The first time they click the "Verify" button in the Client details page and see the icon pop up, they'll be used to seeing that going forward.

Another option would be to show a greyed-out icon or a different icon, but that might clutter up the table view.

**Comment by bmanifold** [comment]
Yeah, I can understand where proactively saying `unverified` could be off putting to some admins.  I could also see some admins potentially wanting to see it to be able to easily know whether they have verified a given device.

I think the `unknown` status could be a good compromise.  I feel like having the row in the details be consistently there is probably a good thing.

**Comment by jamilbk** [comment]
@bmanifold SGTM. Let's roll with that then - just change the label to `Unknown` instead of `unverified`.

**Comment by jamilbk** [comment]
So many things come down to naming...

**Comment by bmanifold** [comment]
> So many things come down to naming...

Indeed.  It's played a much bigger part of my career than I ever would have imagined when I was in school. 😄

**Comment by AndrewDryga** [comment]
What if we just call them `not verified`? `Unknown` looks ambiguous, why our client status is unknown for our own system?.. We also have a bit of space in that vertical table, so even can explain a bit more that not verified means no access if there is a policy with verification condition.

**Comment by jamilbk** [comment]
It's about the device though. Is the device verified? What does it mean to verify a client? I don't think admins are thinking in terms of clients, right? They just want to make sure an unknown device can't connect to their Resources.

Technically the client is verified to have come from Firezone, Inc. because it is signed.

What could make sense then is to move this field to device attributes and call it `device_verification` with status `not verified`. Then in the docs we can explain what this means. Bear in mind though that other systems (notably MDM) will have their _own_ `device_verification` status that can disagree with ours. In that case, `Not verified` contradicts the MDM verification and we would want to bridge the two.

In other words, the usability problem evolves into, "how do I verify my devices?" and the answer of "clicking verify" on each client page doesn't make for a great solution, especially if you consider that the admin actually needs to match the Hardware ID reported by all 100 or so end-users to manually match them up to make sure it actually is the device they're expecting.

We could accept a CSV or bulk upload of known-good hardware or device serials to match against in the policy and call this feature `auto-verification` or something. I see that coming soon due to the usability issues above. I think most orgs would keep an asset list of this sort in their IT system.

**Comment by thomaseizinger** [comment]
What about calling it "New" as in "It is a new device, not seen before"?

**Comment by AndrewDryga** [comment]
WDYT?

<img width="1728" alt="Screenshot 2024-10-11 at 8 32 26 PM" src="https://github.com/user-attachments/assets/15fbcfaf-a324-4325-92d3-e6ab4c9a6781">

> We could accept a CSV or bulk upload of known-good hardware or device serials to match against in the policy and call this feature auto-verification or something. I see that coming soon due to the usability issues above. I think most orgs would keep an asset list of this sort in their IT system.

Or we build MDM integrations. So far nobody was asking for it, device verifications right now is a minimal feature which allows companies to be compliant without having an MDM, I think we should just keep it as that until somebody is willing to pay for us to develop it further.

> What about calling it "New" as in "It is a new device, not seen before"?

It can be manually de-verified by an admin, that's an extra status then, making things worse.

**Comment by AndrewDryga** [comment]
This one looks even better:
<img width="1728" alt="Screenshot 2024-10-11 at 8 36 16 PM" src="https://github.com/user-attachments/assets/b9d77a6a-343d-44ee-8d2d-b6a96c0e335e">

**Comment by jamilbk** [comment]
> This one looks even better: 

LGTM
